### PR TITLE
Issue #4395: increase coverage of pitest-checkstyle-utils profile to 49%

### PR DESCRIPTION
**Pull request by Nimfadora**
Issue #4395 

increased mutation threshold for pitest-checkstyle-common by 9%
threshold: 49%
execution time: 01:36 min
[pitest report before changes for class](https://nimfadora.github.io/com.puppycrawl.tools.checkstyle.utils/ScopeUtils.java.html)
[pitest report before changes for package](https://nimfadora.github.io/issue4395.html)

### CODE REVIEW

**Comment by romani** [comment]
@Nimfadora , Travis is failed - please keep running `mvn verify` before push.

What version of IDEA do you use ?

**Comment by Nimfadora** [comment]
yes I know, it is because I haven't pulled checkstyle/master. I use 2016.3.4 IDEA version. Now everything should be good.

**Comment by romani** [comment]
@Nimfadora , please update to latest IDEA - 2017.1.3. TC is using 2017.1.2 for inspections.

screenshot-idea-violations


screenshot-tc-violations

**Comment by Nimfadora** [comment]
@romani ok, thank you.

File: src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java [hunk 1]
```diff
@@ -99,6 +100,17 @@ public void testInEnumBlockInClass() {
     }
 
     @Test
+    public void testInEnumBlockInLiteralNew() {
+        final DetailAST ast = new DetailAST();
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java]: Why don't we use `getNode`?

File: src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java [hunk 2]
```diff
@@ -132,6 +144,17 @@ public void testIsOuterMostTypeEnum() {
     }
 
     @Test
+    public void testIsOuterMostTypeClass() {
+        final DetailAST ast = new DetailAST();
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java]: same here

File: src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java [hunk 3]
```diff
@@ -149,4 +172,124 @@ public void testIsLocalVariableDefUnexpected() {
...
+    public void testIsInScope() {
+        Assert.assertTrue(ScopeUtils.isInScope(getNodeWithParentScope(TokenTypes.LITERAL_PUBLIC,
+                "public", TokenTypes.ANNOTATION_DEF), Scope.PUBLIC));
+        Assert.assertFalse(ScopeUtils.isInScope(getNodeWithParentScope(TokenTypes.LITERAL_PROTECTED,
+                "protected", TokenTypes.INTERFACE_DEF), Scope.PRIVATE));
+    }
+
+    @Test
+    public void testSurroundingScopeOfNodeChildOfLiteralNewIsAnoninner() {
+        final DetailAST ast = new DetailAST();
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java]: same here

File: src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java [hunk 4]
```diff
@@ -149,4 +172,124 @@ public void testIsLocalVariableDefUnexpected() {
 
         Assert.assertFalse(ScopeUtils.isLocalVariableDef(ast));
     }
+
+    @Test
+    public void testisClassFieldDef() {
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java]: `testis` => `testIs`

File: src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java [hunk 5]
```diff
@@ -149,4 +172,124 @@ public void testIsLocalVariableDefUnexpected() {
...
+        ast2.addChild(ast3);
+        final DetailAST ast4 = new DetailAST();
+        ast4.setType(TokenTypes.MODIFIERS);
+        ast3.addChild(ast4);
+        Assert.assertTrue(ScopeUtils.isInAnnotationBlock(ast3));
+        Assert.assertFalse(ScopeUtils.isInAnnotationBlock(ast4));
+    }
+
+    @Test
+    public void testisInInterfaceOrAnnotationBlock() {
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java]: `testis` => `testIs`

File: src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java [hunk 6]
```diff
@@ -149,4 +172,124 @@ public void testIsLocalVariableDefUnexpected() {
...
+                getNode(TokenTypes.INTERFACE_DEF, TokenTypes.OBJBLOCK)));
+        Assert.assertFalse(ScopeUtils.isInInterfaceOrAnnotationBlock(
+                getNode(TokenTypes.CLASS_DEF, TokenTypes.OBJBLOCK)));
+        Assert.assertFalse(ScopeUtils.isInInterfaceOrAnnotationBlock(
+                getNode(TokenTypes.LITERAL_NEW, TokenTypes.IDENT)));
+        Assert.assertFalse(ScopeUtils.isInInterfaceOrAnnotationBlock(
+                getNode(TokenTypes.ENUM_DEF, TokenTypes.OBJBLOCK)));
+    }
+
+    private static DetailAST getNode(int parentType, int nodeType) {
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java]: Could we make this `getNode(int... nodeTypes)` and use it in more places?

File: src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java [hunk 7]
```diff
@@ -149,4 +172,124 @@ public void testIsLocalVariableDefUnexpected() {
...
+        ast.setType(parentType);
+        final DetailAST ast2 = new DetailAST();
+        ast2.setType(nodeType);
+        ast.addChild(ast2);
+        return ast2;
+    }
+
+    private static DetailAST getNodeWithParentScope(int literal, String scope,
+                                                    int parentTokenType) {
+        final DetailAST ast1 = new DetailAST();
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java]: In every other test, you name variables `ast`, `ast2`, etc. and you put them in order of variable names. This is the only one that breaks that pattern.
Can we make the layout here similar?

File: src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java [hunk 8]
```diff
@@ -149,4 +172,124 @@ public void testIsLocalVariableDefUnexpected() {
...
+    }
+
+    @Test
+    public void testSurroundingScopeOfNodeChildOfLiteralNewIsAnoninner() {
+        final DetailAST ast = new DetailAST();
+        ast.setType(TokenTypes.LITERAL_NEW);
+        final DetailAST ast2 = new DetailAST();
+        ast2.setType(TokenTypes.IDENT);
+        ast.addChild(ast2);
+        Assert.assertEquals(Scope.ANONINNER, ScopeUtils.getSurroundingScope(ast2));
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java]: Please separate assert from previous statements like you did for other tests.

File: src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java [hunk 9]
```diff
@@ -149,4 +172,124 @@ public void testIsLocalVariableDefUnexpected() {
...
+        final DetailAST ast2 = new DetailAST();
+        ast2.setType(TokenTypes.OBJBLOCK);
+        ast.addChild(ast2);
+        final DetailAST ast3 = new DetailAST();
+        ast3.setType(TokenTypes.CLASS_DEF);
+        ast2.addChild(ast3);
+        final DetailAST ast4 = new DetailAST();
+        ast4.setType(TokenTypes.MODIFIERS);
+        ast3.addChild(ast4);
+        Assert.assertTrue(ScopeUtils.isInInterfaceBlock(ast3));
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java]: same here.

File: src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java [hunk 10]
```diff
@@ -149,4 +172,124 @@ public void testIsLocalVariableDefUnexpected() {
...
+        final DetailAST ast2 = new DetailAST();
+        ast2.setType(TokenTypes.OBJBLOCK);
+        ast.addChild(ast2);
+        final DetailAST ast3 = new DetailAST();
+        ast3.setType(TokenTypes.INTERFACE_DEF);
+        ast2.addChild(ast3);
+        final DetailAST ast4 = new DetailAST();
+        ast4.setType(TokenTypes.MODIFIERS);
+        ast3.addChild(ast4);
+        Assert.assertTrue(ScopeUtils.isInAnnotationBlock(ast3));
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java]: same here.

**Comment by rnveach** [comment]
@Nimfadora Why did we close this?

**Comment by Nimfadora** [comment]
@rnveach, wow, It has happened accidentally, I suppose. Don't know how, maybe I have pushed the button by mistake. Can we reopen it somehow?

**Comment by Nimfadora** [comment]
Closing this PR was a mistake, reopening.

File: src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java (hunk 1 above)
Comment (Nimfadora) [inline src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java]: Fixed

File: src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java (hunk 2 above)
Comment (Nimfadora) [inline src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java]: Fixed

File: src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java (hunk 4 above)
Comment (Nimfadora) [inline src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java]: Fixed

File: src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java (hunk 3 above)
Comment (Nimfadora) [inline src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java]: Fixed

File: src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java (hunk 8 above)
Comment (Nimfadora) [inline src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java]: Fixed

File: src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java (hunk 9 above)
Comment (Nimfadora) [inline src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java]: Fixed

File: src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java (hunk 10 above)
Comment (Nimfadora) [inline src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java]: Fixed

File: src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java (hunk 5 above)
Comment (Nimfadora) [inline src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java]: Fixed

File: src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java (hunk 6 above)
Comment (Nimfadora) [inline src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java]: Done

File: src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java (hunk 7 above)
Comment (Nimfadora) [inline src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java]: Fixed

File: src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java [hunk 11]
```diff
@@ -36,117 +39,163 @@ public void testIsProperUtilsClass() throws ReflectiveOperationException {
...
+        }
+        return ast;
+    }
 
-        Assert.assertFalse(ScopeUtils.isLocalVariableDef(ast));
+    private static DetailAST getNodeWithParentScope(int literal, String scope,
+                                                    int parentTokenType) {
+        final DetailAST ast = getNode(parentTokenType, TokenTypes.MODIFIERS, literal);
+        ast.setText(scope);
+        final DetailAST ast2 = getNode(TokenTypes.OBJBLOCK);
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java]: it is first time when codacy-bot is right and nobody else catch a problem.
@Nimfadora , please fix.

**Comment by Nimfadora** [comment]
@romani, I have reinstalled idea, now it is of 2017.1.3 version. Configured profile and scope, run inspections, and unfortunately I can't reproduce this issue with method name length. 
It seems that `JUnit 4+ test method naming convention` overrides `Instance method naming convention`. Cause I've got an error when method is more than 80 characters length.

File: src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java (hunk 11 above)
Comment (Nimfadora) [inline src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java]: Fixed

**Comment by romani** [comment]
@Nimfadora ,

> Cause I've got an error when method is more than 80 characters length.

it mean that you still have mis-configured IDEA as it works on my side the same as on TC. There should be no warnings/errors at all from inspection. If any present - you use different scope or profile.
Please recheck one more time.

**Comment by rnveach** [comment]
@romani feel free to merge if you are ready

**Comment by romani** [comment]
@Nimfadora , issue with IDEA setup still on you, it has to be resolved, recheck all, if still no success, find me in gitter for discussion

**Pull request merged**